hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -205,6 +210,15 @@ public <I, O> O aggregate(HoodieData<I> data, O zeroValue, Functions.Function2<O
     return data.collectAsList().stream().parallel().reduce(zeroValue, seqOp::apply, combOp::apply);
   }
 
+  @Override
+  public ReaderContextFactory<?> getReaderContextFactory(HoodieTableMetaClient metaClient) {
+    // metadata table and HFile reads are only supported by the AvroReaderContext.
+    if (metaClient.isMetadataTable() || HFILE.getFileExtension().equals(metaClient.getTableConfig().getBaseFileFormat().getFileExtension())) {
```

Comment:
in which case the data table cound have `HFile` as the base file format.

Reply:
I don't think it is supported, the test should be aming to test the MDT instead?

Reply:
The test is `TestHoodieSparkMergeOnReadTableInsertUpdateDelete#testSimpleInsertAndUpdate`, the case for HFile will need to be removed if we remove this check

Reply:
yeah, let's update the test.